### PR TITLE
chore: bump addon and gatekeeper 3.17

### PR DESCRIPTION
**Pull request by kevinhwangkr-msft**
**What type of PR is this?**

bump add on and gatekeeper 3.17 image 

/kind cleanup

**What this PR does / why we need it**:

to bump the add on and gatekeeper version

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:

- [X] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:

```
none
```

### CODE REVIEW

File: parts/linux/cloud-init/artifacts/components.json [hunk 1]
```diff
@@ -332,6 +334,7 @@
       "downloadURL": "mcr.microsoft.com/azure-policy/policy-kubernetes-webhook:*",
       "amd64OnlyVersions": [],
       "multiArchVersions": [
+        "1.7.1",
         "1.7.0",
         "1.5.0"
```
Comment (cameronmeissner) [inline parts/linux/cloud-init/artifacts/components.json:339]: can we remove 1.5.0?

Comment (kevinhwangkr-msft) [reply parts/linux/cloud-init/artifacts/components.json:339]: 1.5.0 is still used and will be supported until next year

Please confirm @anlandu

Comment (anlandu) [reply parts/linux/cloud-init/artifacts/components.json:339]: 1.5 was supposed to be fully switched over to 1.7 in RP release 0827 but that got paused so we still have quite a few clusters on 1.5--will remove ASAP but ideally this PR goes out before 1.5 removal as 1.7.1 patches a CVE

**Review by cameronmeissner — APPROVED** [review]
cve fix, 1.5 will be removed ASAP

**Comment by ganeshkumarashok** [comment]
Can 1.5 be removed now?

**Pull request merged**